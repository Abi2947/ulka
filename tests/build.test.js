require('colors')
const fs = require('fs')
const build = require('../bin/build')
const absolutePath = require('../src/utils/absolutePath')
const globalInfo = require('../src/globalInfo')

const curDir = process.cwd()

describe('build function: ', () => {
  beforeEach(() => {
    process.chdir(absolutePath('tests/resources/ulka_test'))
  })

  afterEach(() => {
    process.chdir(curDir)
  })

  test('globalInfo should have info about content files', async () => {
    await build()
    const contentFiles = globalInfo.contentFiles.map(f => {
      delete f.createFilePath
      delete f.link
      return f
    })

    expect(contentFiles).toMatchSnapshot()
  })

  test('bulid folder should be created', () => {
    const buildExists = fs.existsSync(globalInfo.configs.buildPath)
    expect(buildExists).toBe(true)
  })
})
