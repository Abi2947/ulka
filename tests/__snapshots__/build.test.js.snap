// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build function:  globalInfo should have info about content files 1`] = `
Array [
  Object {
    "absoluteFilePath": "C:\\\\Users\\\\rosha\\\\Desktop\\\\ulka-org\\\\ulka\\\\tests\\\\resources\\\\ulka_test\\\\build\\\\blog\\\\post-1\\\\index.html",
    "contentsName": "contents",
    "frontMatter": Object {
      "title": "Post 1",
    },
    "html": "<html>
    <head>
        <title>Ulka Js</title>
    </head>
    <body>
        <h1>This is post one</h1>
    </body>
</html>",
  },
  Object {
    "absoluteFilePath": "C:\\\\Users\\\\rosha\\\\Desktop\\\\ulka-org\\\\ulka\\\\tests\\\\resources\\\\ulka_test\\\\build\\\\blog\\\\post-2\\\\index.html",
    "contentsName": "contents",
    "frontMatter": Object {
      "title": "Post 2",
    },
    "html": "<html>
    <head>
        <title>Ulka Js</title>
    </head>
    <body>
        <h1>Post 2</h1>
    </body>
</html>",
  },
]
`;
